git/index/base.py

```diff
@@ -351,7 +351,7 @@ def from_tree(cls, repo: "Repo", *treeish: Treeish, **kwargs: Any) -> "IndexFile
 
         # tmp file created in git home directory to be sure renaming
         # works - /tmp/ dirs could be on another device
-        tmp_index = tempfile.mktemp("", "", repo.git_dir)
+        tmp_index = tempfile.NamedTemporaryFile(dir=repo.git_dir, delete=False).name
```

Comment:
I think this effectively opens a temporary file, extracts the name, and then forgets about the opened file.
As there are no destructors in `python`, it's unclear when (or if) this resource is ever freed and is effectively leaking a very limited process resource over time.